### ISSUE TITLE
Need a way to specify Projects to build in repo specific targets file

### ISSUE THREAD

**Issue by weshaggard** [dotnet/arcade#761]
Today the build.ps1/sh scripts expect the projects to build are a sln file in the root of the repo for the core repo's that isn't the case. The script files allow you to pass in a `-projects` parameter to override the project but that causes issues because we now have to pass it in all the time for any commands to build or cibuild which adds an extra layer of complexity. Instead we should be able to specify the projects in a targets/props file under the 'eng' directory that our build.proj will import and pick-up. 

@tmat what is your thoughts about having ReproLayout.props import a similarly named file under the eng directory if it exists. That allows the repo's to customize some of these global properties if necessary.

**Comment by tmat** [comment]
Makes sense. This would also allow to customize what projects to build based on an arbitrary condition (including, e.g. `DotNetBuildFromSource`).
